cores/arduino/wiring_analog.c

```diff
@@ -61,6 +61,21 @@ uint32_t analogRead(uint32_t ulPin)
   return value;
 }
 
+//Start a DMA acquisition on ADC
+//the initialization of the analog PIN is done through this function
+void analogReadDma( uint32_t ulPin, uint32_t *pData, uint32_t lData, void (*callback)(void *user_data))
```

Comment:
if we use a user_data parameter in the callback, then the user data paramrter must also be passed by application.